juniper/src/executor/mod.rs

```diff
@@ -666,31 +667,30 @@ where
                     None
                 }
             })
-            .filter(|s| s.is_some())
+            .flatten()
```

Comment:
This change wasn't strictly necessary but it's (as far as I can tell) equivalent, and IMO makes it a bit easier to read. Previously the `.filter` was doing this mapping:
```
None => None,
Some(None) => None,
Some(Some(x)) => Some(Some(x))
```

and now it's doing:

```
None => None,
Some(None) => None,
Some(Some(x)) => Some(x)
```

so the second layer of `Option` was redundant, so the `flatten` removes that